### PR TITLE
Resolve collision between WooPayments header and Woo Express survey banner

### LINKED ISSUE #8093: WooPayments header collides with Woo Express survey banner.
### Describe the bug
When a site is hosted on Woo Express, store owners typically see the Woo Express survey header at the top of the page. However, an issue arises when a user clicks an action that redirects to the WooPayments setup wizard. In this scenario, the WooPayments header appears, leading to a collision where the Woo Express Survey header is obscured by the WooPayments header.

### To Reproduce

1. Start Woo Express trial
2. Make sure that you don't finish Woo Express survey. It should be on the top of the page.
3. Once the initial setup is finished, click on the action item to start with setting up WooPayments
4. On the wizard page you will notice Woo Express survey header behind WooPayments header (check screenshot below)

### Actual behavior
Woo Express survey appearing behind the Woo Payments header

### Screenshots
image

### Expected behavior
Woo Express survey and Woo Payments header should not collide.

### Desktop (please complete the following information):

* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

### Smartphone (please complete the following information):
N/A

### Additional context
More info can be found here: peapX7-45b-p2

### PR DESCRIPTION
**Pull request by mordeth**
Fixes #8093

#### Changes proposed in this Pull Request
This PR removes the JITM notices on onboarding page to prevent any distractions during onboarding.

#### Testing instructions
- Start Woo Express trial.
- Make sure that you don't finish Woo Express survey. It should be on the top of the page.
- Once the initial setup is finished, click on the action item to start with setting up WooPayments.
- Upload WooPayment plugin with current change.
- Use the following [gist].(https://gist.githubusercontent.com/ilyasfoo/a2901efacd62f69be737778ba7721e69/raw/5e18c6fc82f52ecf4c552cec5bc3b092540338dd/jQueryJITM) to force show the survey banner.
- Verify on the wizard page the survey banner is hidden.

*

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8400` or branch name `fix/8093-onboarding-collides-with-jitm-banners` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/8093-onboarding-collides-with-jitm-banners) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 5df331f1687d000e81538a4dbbcd6a3f5b68ceb0
- Build time: 2024-03-28 12:03:12 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/onboarding/style.scss [hunk 1]
```diff
@@ -168,4 +168,9 @@ body.wcpay-onboarding__body {
 			color: $gray-700;
 		}
 	}
+
+	// Reposition JITM banners below the fixed onboarding header.
```
Comment (elazzabi) [inline client/onboarding/style.scss]: Just to make it clearer 😄 
```suggestion
	// Reposition Jetpack's JITM (Just in time messages) banners below the fixed onboarding header.
```

**Review by elazzabi — CHANGES_REQUESTED** [review]
Left a small request to make the comment clearer.

Also, for my curiosity, I asked on Slack for more details about this decision p1710986366846829/1710986315.983159-slack-C03KTTK2YMA

**Review by elazzabi — APPROVED** [review]
After the discussion on Slack, it's up to the team to decide what's the best approach here. Removing my block to merge if we decide to do so.

**Comment by elizaan36** [comment]
imo this survey should be removed. There shouldn't be any competing calls to action during WooPayments onboarding, especially non-essential ones like this. It's a distraction that can impact activation.

**Comment by vladolaru** [comment]
> imo this survey should be removed. There shouldn't be any competing calls to action during WooPayments onboarding, especially non-essential ones like this. It's a distraction that can impact activation.

Totally agree, @elizaan36

**Comment by mordeth** [comment]
I've revised the initial approach and eliminated the JITM notices entirely from the onboarding page to ensure there are no distractions.

**Review by oaratovskyi — APPROVED** [review]
It was tricky to test, but it works! :shipit: 
Testing instructions are slightly different, see p1711623868207219-slack-C03KTTK2YMA

**Review by elazzabi — APPROVED** [review]
Removing it completely looks like a better choice! Approving :shipit:

**Pull request merged**